tests/integration-service/README.md

```diff
@@ -0,0 +1,98 @@
+# Integration Service
+
+The Integration Service manages the lifecycle, deployment and integration testing of applications and components. It handles snapshots, environments, pipelines and deployments, ensuring applications run efficiently and are deployed correctly.
+
+
+## Happy Path Tests.
+
+Happy path testing describes tests that focus on the most common scenarios while assuming there are no exceptions or errors.
+### Basic Integration Tests:
+
+- Testing for successful creation of applications and components.
+- Asserting the successful creation of a snapshot after a push event.
+- Checking if the BuildPipelineRun is successfully triggered and completed.
+- Asserting the signing of BuildPipelineRun.
+- Validating the successful creation of a SnapshotEnvironmentBinding.
+- Validating the successful creation of a Release.
+- Validating the Global Candidate is updated
+- Verifying that all integration pipeline runs finish successfully.
+- Checking that a snapshot gets successfully marked as 'passed' after all Integration pipeline runs finished.
+
+## Negative Test Cases.
+
+### Failed IntegrationTestScenarios:
+
+- Creating an IntegrationTestScenario that is expected to fail.
+- Asserting that a snapshot is marked as failed.
+- Validating that no Release CRs and no SnapshotEnvironmentBinding are created in certain scenarios.
+- Checking that the global candidate does not get updated unexpectedly.
+
+
+### Failed Ephemeral Environment provisioning due to missing DeploymentTargetClass:
+- Verifying that entities like deploymentTargetClass and GitOpsCR don't exist under certain conditions.
+- Asserting that no GitOpsCR is created for a non-existing deploymentTargetClass.
+- Checking that a snapshot doesn't get marked as 'passed' under specific conditions.
+  
+
+# Running e2e tests.
+
+### 1. Log in to Openshift cluster with OpenShift command line tool as admin (default is kubeadmin)
+
+```bash
+oc login -u <user> -p <password> --server=<oc_api_url>
+```
+
+### 2. Export environment variables from default.env file as follows:
+
+```bash
+$ cp default.env user.env
+```
+
+Run source command on the user.env file. This file will need to be sourced each time you log into a new cluster.
+
+```bash
+$ source user.env
+```
+
+### 3. Install dependencies:
+
+```bash
+$ go mod tidy
+# or go mod tidy -compat=1.19
+# Copy the dependencies to vendor folder
+$ go mod vendor
+```
+
+### 4. Build images:
+
+```bash
+$ make build
+```
+
+### 5. Install infra-deployments:
+
+```bash
+$ make local/cluster/prepare
+```
```

Comment:
These instructions are already present here: https://github.com/redhat-appstudio/e2e-tests#install-appstudio-in-e2e-mode, do we still need to mention them here again?

What do we think about this?

Reply:
I would suggest to provide a link to that section of the documentation.